File: FashionShop.Web/src/features/products/types/product.ts
export interface Product {
    id: string;
    categoryId: string;
    brandId: string;
    categoryName: string;
    brandName: string;
    name: string;
    slug: string;
    description: string;
    content: string;
    material: string;
    price: number;
    thumbnailUrl: string;
    isActive: boolean;
    isBestSeller: boolean;
    isNew: boolean;
    viewCount: number;
    createdDate: Date;
    updatedDate: Date;
}

export interface ProductVariant {
    id: string;
    productId: string;
    colorId: string;
    sizeId: string;
    sku: string;
    quantity: number;
    price: number;
    isActive: boolean;
    createdDate: Date;
    updatedDate: Date;
}

export interface ProductDetail extends Product {
    productVariants: ProductVariant[];
}

File: FashionShop.Backend/FashionShop.Core/Entities/Product.cs
using FashionShop.Core.Common;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FashionShop.Core.Entities
{
    [Table("Products")]
    public class Product : BaseEntity
    {
        [Required]
        [MaxLength(200)]
        public string Name { get; set; }

        [Required]
        [MaxLength(200)]
        [Column(TypeName = "varchar(200)")]
        public string Slug { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        [MaxLength(100)]
        public string Material { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        [Required]
        [MaxLength(500)]
        [Column(TypeName = "varchar(500)")]
        public string ThumbnailUrl { get; set; }

        [Required]
        public Guid CategoryId { get; set; }

        [Required]
        public Guid BrandId { get; set; }
        public bool IsActive { get; set; } = true;
        public bool IsBestSeller { get; set; } = false;
        public bool IsNew { get; set; } = true;
        public int ViewCount { get; set; } = 0;

        [ForeignKey("CategoryId")]
        public virtual Category Category { get; set; }

        [ForeignKey("BrandId")]
        public virtual Brand Brand { get; set; }

        public virtual ICollection<ProductVariant> ProductVariants { get; set; } = new List<ProductVariant>();

        public virtual ICollection<ProductImage> ProductImages { get; set; } = new List<ProductImage>();

        public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
    }
}
